solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/realtime/RealtimeSolrSearchSink.java

```diff
@@ -0,0 +1,199 @@
+        } else {
+          document.addField(solrFieldName, structuredRecord.get(solrFieldName));
+        }
+      }
+      documentList.add(document);
+      numRecordsWritten++;
+    }
+    solrClient.add(documentList);
+    solrClient.commit();
+    return numRecordsWritten;
+  }
+
+  @Override
+  public void destroy() {
+    solrClient.shutdown();
+  }
+
+  /**
+   * Returns the SolrClient for establishing the connection with Solr server, using the hostname and port provided by
+   * user.
+   *
+   * @return Solr client
+   */
+  private SolrClient getSolrConnection() {
+    String urlString;
+    SolrClient solrClient = null;
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE)) {
+      urlString = "http://" + config.getSolrHost() + "/solr/" + config.getCollectionName();
+      solrClient = new HttpSolrClient(urlString);
+    } else if (config.getSolrMode().equals(SOLR_CLOUD_MODE)) {
+      CloudSolrClient solrCloudClient = new CloudSolrClient(config.getSolrHost());
+      solrCloudClient.setDefaultCollection(config.getCollectionName());
+      solrClient = solrCloudClient;
+    }
+    return solrClient;
+  }
+
+  /**
+   * Validates whether the host entered for Single Node Solr instance is proper or not.
+   */
+  private void validateSolrConnectionString() {
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE) && config.getSolrHost().contains(",")) {
+      throw new IllegalArgumentException(String.format("Multiple hosts '%s' found for Single Node Solr.",
+                                                       config.getSolrHost()));
+    }
+  }
+
+  /**
+   * Verifies whether the Solr server is running or not. Also, validates whether the collection provided by user is
+   * registered with the Solr server or not.
+   */
+  private void verifySolrConfiguration() {
+    SolrClient client = getSolrConnection();
+    try {
+      client.ping();
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+        String.format("Server refused connection at '%s'. Please make sure that either the Solr/Zookeeper services " +
+                        "are properly running or the collection '%s' exists in the Solr Server.", config.getSolrHost(),
+                      config.getCollectionName()));
+    }
+  }
+
+  /**
+   * Validates whether the Id field provided by user is present in the input schema or not.
+   *
+   * @param inputSchema
+   */
+  private void validateIdField(Schema inputSchema) {
+    if (inputSchema != null && inputSchema.getField(config.getIdField()) == null) {
+      throw new IllegalArgumentException(
+        String.format("Idfield '%s' does not exist in the input schema %s", config.getIdField(), inputSchema));
+    }
+  }
+
+  /**
+   * Validates whether the CDAP data types coming from the input schema are compatible with the Solr data types or not.
+   *
+   * @param inputSchema
+   */
+  private void validateInputFieldsDataType(Schema inputSchema) {
```

Comment:
seems like we can refactor some of the methods from this class and the batch sink. The implementation of some methods seem to be duplicate.
Consider a Solr utility class, which both sinks can use.